app/src/main/cpp/skyline/soc/gm20b/engines/maxwell_dma.cpp

```diff
@@ -30,167 +30,160 @@ namespace skyline::soc::gm20b::engine {
     }
 
     void MaxwellDma::LaunchDma() {
-        if (*registers.lineLengthIn == 0)
-            return; // Nothing to copy
+        if (registers.launchDma->remapEnable) [[unlikely]]
+            Logger::Warn("Remapped DMA copies are unimplemented!");
+        else
+            DmaCopy();
 
-        if (registers.launchDma->remapEnable) {
-            Logger::Warn("DMA remapping is unimplemented!");
-            return;
-        }
+        ReleaseSemaphore();
+    }
+
+    void MaxwellDma::DmaCopy() {
 
         if (registers.launchDma->multiLineEnable) {
             channelCtx.executor.Submit();
-            if (registers.launchDma->srcMemoryLayout == Registers::LaunchDma::MemoryLayout::Pitch &&
-                registers.launchDma->dstMemoryLayout == Registers::LaunchDma::MemoryLayout::BlockLinear)
-                CopyPitchToBlockLinear();
-            else if (registers.launchDma->srcMemoryLayout == Registers::LaunchDma::MemoryLayout::BlockLinear &&
-                registers.launchDma->dstMemoryLayout == Registers::LaunchDma::MemoryLayout::Pitch)
+
+            if (registers.launchDma->srcMemoryLayout == registers.launchDma->dstMemoryLayout) [[unlikely]] {
+                // Pitch to Pitch copy
+                if (registers.launchDma->srcMemoryLayout == Registers::LaunchDma::MemoryLayout::Pitch) {
+                    if ((*registers.pitchIn == *registers.pitchOut) && (*registers.pitchIn == *registers.lineLengthIn)) {
+                        // Both Linear...copy as is.
```

Comment:
```suggestion
                        // Both Linear, copy as is
```
We tend to not terminate comments with full-stops in Skyline